servers/rendering/renderer_rd/storage_rd/material_storage.cpp

```diff
@@ -2379,6 +2379,9 @@ MaterialStorage::Samplers MaterialStorage::samplers_rd_allocate(float p_mipmap_b
 	RD::SamplerFilter mip_filter = samplers.use_nearest_mipmap_filter ? RD::SAMPLER_FILTER_NEAREST : RD::SAMPLER_FILTER_LINEAR;
 	float anisotropy_max = float(1 << samplers.anisotropic_filtering_level);
 
+	// Intel GPUs don't currently support anisotropic filtering with nearest-neighbor
+	const bool is_intel_gpu = RD::get_singleton()->get_device_vendor_name() == "Intel";
```

Comment:
This check should be moved in the class constructor (it should set a member variable), so it's performed only once on startup.